### PR TITLE
Major refactoring to monitor

### PR DESCRIPTION
**Pull request by josephschorr**
A major refactoring of the monitor to remove AWS metrics, add Prometheus metrics, and fix the monitor to work properly when run itself under Docker

### CODE REVIEW

File: monitor.go [hunk 1]
```diff
@@ -177,50 +202,83 @@ func clearAllImages(dockerClient *docker.Client) bool {
...
+			if _, toSkip := skipImages[image.ID]; toSkip {
+				continue
+			}
+
+			log.Infof("Clearing image %s", image.ID)
 			if err = dockerClient.RemoveImage(image.ID); err != nil {
-				log.Printf("%s", err)
-				healthy = false
-				return false
+				if os.Getenv("UNDER_DOCKER") != "true" {
```
Comment (jzelinskie) [inline monitor.go]: Lines like these should be:
```go
strings.ToLower(os.Getenv("UNDER_DOCKER") != "true"
```

**Comment by josephschorr** [comment]
@jzelinskie: I like the status handler since it can be used with a normal ELB or K8s service check.

**Comment by jzelinskie** [comment]
A ping web service check should be checking the health of the monitor and not dealing with logic related to it's pushes and pulls.

**Comment by josephschorr** [comment]
Health of the monitor is for verifying the monitor is up and running; status is for verifying the *registry* is up and running. For example, pingdom hits our `/status` right now.

**Comment by jzelinskie** [comment]
Yeah, pingdom shouldn't do that. There should be a prom alert on failure counter.

**Comment by josephschorr** [comment]
But if the user isn't using prometheus, the monitor still has value. Prometheus is optional in my opinion.

File: monitor.go (hunk 1 above)
Comment (josephschorr) [inline monitor.go]: Fixed

**Pull request merged**